### PR TITLE
Move logic for calling S3 select to a shared function

### PR DESCRIPTION
**Pull request by kevinemoore**
## Description

Multiple lambdas will use s3 select to read manifests in s3 so move querying logic into a single shared function in t4_lambda_shared.

### CODE REVIEW

File: lambdas/shared/t4_lambda_shared/utils.py [hunk 1]
```diff
@@ -105,3 +105,30 @@ def buffer_s3response(s3response):
         raise IncompleteResultException("Error: Received an incomplete response from S3 Select.")
     response.seek(0)
     return response
+
+
+def call_s3_select(
```
Comment (akarve) [inline lambdas/shared/t4_lambda_shared/utils.py:110]: `select_from_manifest`

File: lambdas/shared/t4_lambda_shared/utils.py [hunk 2]
```diff
@@ -105,3 +105,30 @@ def buffer_s3response(s3response):
...
+    key: str,
+    sql_stmt: str
+) -> io.StringIO:
+    """
+    Call S3 Select to read only the logical keys from a
+    package manifest that match the desired folder path
+    prefix
+    """
+
+    print(sql_stmt)
```
Comment (akarve) [inline lambdas/shared/t4_lambda_shared/utils.py:122]: In prod?

File: lambdas/shared/t4_lambda_shared/utils.py [hunk 3]
```diff
@@ -105,3 +105,30 @@ def buffer_s3response(s3response):
         raise IncompleteResultException("Error: Received an incomplete response from S3 Select.")
     response.seek(0)
     return response
+
+
+def call_s3_select(
+    s3_client: str,
```
Comment (akarve) [inline lambdas/shared/t4_lambda_shared/utils.py:111]: You could do only s3_client as positional, or s3_client, bucket, and key as positional, the rest should be required kwargs so:

```
s3_client: str,
*,
bucket
key,
etc,
```

File: lambdas/shared/t4_lambda_shared/utils.py (hunk 2 above)
Comment (kevinemoore) [reply lambdas/shared/t4_lambda_shared/utils.py:122]: Yes. That will be very helpful in the lambda logs, unless there's a better way to log the SQL?

Comment (akarve) [inline lambdas/shared/t4_lambda_shared/utils.py:122]: Fine but you really need to give this some context:
```suggestion
    print(f"utils.py: manifest_select: {sql_stmt})
```

File: lambdas/pkgselect/index.py [hunk 4]
```diff
@@ -72,47 +72,13 @@ def get_s3_client(
...
 
 
-def call_s3_select(
-    s3_client: str,
-    bucket: str,
-    key: str,
-    logical_key_prefix: str,
-    detail=False
-):
+def fix_quotes(key: str):
```
Comment (akarve) [inline lambdas/pkgselect/index.py:75]: Dima already has something for this in access_counts, just factor that out into shared as well:
https://github.com/quiltdata/quilt/blob/master/lambdas/access_counts/index.py#L34

**Pull request merged**